src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -840,6 +846,57 @@ void WindowImplX11::setSize(const Vector2u& size)
     XFlush(m_display);
 }
 
+////////////////////////////////////////////////////////////
+void WindowImplX11::setMinimumSize(const Vector2u &minimumSize)
+{
+    assert(minimumSize.x < m_maximumSize.x);
+    assert(minimumSize.y < m_maximumSize.y);
+
+    // Not sure what to do when resizing is disabled.
+    // Ignore setMinimumSize is what I would recommend
+    if (m_useSizeHints)
+    {
+        assert(false);
+    }
+    else
+    {
+        m_minimumSize = minimumSize;
+        XSizeHints* sizeHints = XAllocSizeHints();
+        sizeHints->flags = PMinSize | PMaxSize;
+        sizeHints->min_width = static_cast<int>(minimumSize.x);
+        sizeHints->min_height = static_cast<int>(minimumSize.y);
+        sizeHints->max_width = static_cast<int>(m_maximumSize.x);
+        sizeHints->max_height = static_cast<int>(m_maximumSize.y);
+        XSetWMNormalHints(m_display, m_window, sizeHints);
+        XFree(sizeHints);
+    }
+}
+
+////////////////////////////////////////////////////////////
+void WindowImplX11::setMaximumSize(const Vector2u &maximumSize)
+{
+    assert(maximumSize.x > m_minimumSize.x);
+    assert(maximumSize.y > m_minimumSize.y);
+
+    // Not sure what to do when resizing is disabled.
+    // Ignore setMaximumSize is what I would recommend
+    if (m_useSizeHints)
+    {
+        assert(false);
+    }
+    else
+    {
+        m_maximumSize = maximumSize;
+        XSizeHints* sizeHints = XAllocSizeHints();
+        sizeHints->flags = PMaxSize | PMinSize;
+        sizeHints->max_width = static_cast<int>(maximumSize.x);
+        sizeHints->max_height = static_cast<int>(maximumSize.y);
+        sizeHints->min_width = static_cast<int>(m_minimumSize.x);
+        sizeHints->min_height = static_cast<int>(m_minimumSize.y);
+        XSetWMNormalHints(m_display, m_window, sizeHints);
+        XFree(sizeHints);
+    }
```

Comment:
maybe, but not yet. Every assert is me signifying undefined behavior that we need to figure out the best way of handling. What should we do if the user decides to set a min window size despite not using a resizable window style.

Reply:
What we could do here:

1. Do nothing if the window style is set to no resize
2. Allow a resize to occur to the min size if the setMinimumSize is called and do the same for the set maximum method.
3. Assert

I think the 2nd option is what I'd go with.

Reply:
Also, they aren't identical

> Consider extracting this mostly-duplicated code to a private function, or -- even better -- a function in an anonymous namespace.

They aren't identical. I don't see how you could abstract this.

Reply:
>They aren't identical.

They can be identical if you use member variables `m_minimumSize` and `m_maximumSize` to setup the `XSizeHints` instance instead of arguments.

Reply:
> > They aren't identical.
> 
> They can be identical if you use member variables `m_minimumSize` and `m_maximumSize` to setup the `XSizeHints` instance instead of arguments.

Oh huh. yeah. ill do that then. thanks.

Reply:
> > Consider extracting this mostly-duplicated code to a private function, or -- even better -- a function in an anonymous namespace.
> 
> They aren't identical. I don't see how you could abstract this with a function.

The parts that differ can be passed as parameters.

```cpp
namespace {

void updateSizeHints(const Vector2u& mins, const Vector2u& maxs, ::Display* display, ::Window window)
{
    XSizeHints* sizeHints = XAllocSizeHints();
    sizeHints->flags = PMinSize | PMaxSize;
    sizeHints->min_width = static_cast<int>(mins.x);
    sizeHints->min_height = static_cast<int>(mins.y);
    sizeHints->max_width = static_cast<int>(maxs.x);
    sizeHints->max_height = static_cast<int>(maxs.y);
    XSetWMNormalHints(display, window, sizeHints);
    XFree(sizeHints);
}

}

////////////////////////////////////////////////////////////
void WindowImplX11::setMinimumSize(const Vector2u &minimumSize)
{
    assert(minimumSize.x <= m_maximumSize.x);
    assert(minimumSize.y <= m_maximumSize.y);

    // Not sure what to do when resizing is disabled.
    // Ignore setMinimumSize is what I would recommend
    if (m_useSizeHints)
    {
        assert(false);
    }
    else
    {
        m_minimumSize = minimumSize;
        updateSizeHints(minimumSize, m_maximumSize, m_display, m_window);
    }
}

////////////////////////////////////////////////////////////
void WindowImplX11::setMaximumSize(const Vector2u &maximumSize)
{
    assert(maximumSize.x >= m_minimumSize.x);
    assert(maximumSize.y >= m_minimumSize.y);

    // Not sure what to do when resizing is disabled.
    // Ignore setMaximumSize is what I would recommend
    if (m_useSizeHints)
    {
        assert(false);
    }
    else
    {
        m_maximumSize = maximumSize;
        updateSizeHints(maximumSize, m_minimumSize, m_display, m_window);
    }
}
```

Reply:
Done. I just didn't think about that for some reason.